font-codegen/src/fields.rs

```diff
@@ -1216,7 +1257,20 @@ impl FieldType {
         }
     }
 
-    fn compile_type(&self, nullable: bool, version_dependent: bool) -> TokenStream {
+    /// The tokens used to represent this type in `write-fonts`.
+    ///
+    /// The `nullable` and `version_dependent` arguments are passed in, because
+    /// they live in the FieldAttrs that are not accessible from FieldType.
+    ///
+    /// The `for_constructor` argument is `true` only we want the type of an
+    /// argument to a constructor method. Specifically, this does not wrap
+    /// offset-types in offset markers (these conversions occur in the constructor).
+    fn compile_type(
+        &self,
+        nullable: bool,
+        version_dependent: bool,
+        for_constructor: bool,
```

Comment:
not loving the bool typing here, it's resulting in code like `self.typ.compile_type(self.is_nullable(), false, true))` where the is_nullable is clear but the literals have non-obvious meaning. Feels like a case where a typed flag (enum?) or builder would result in clearer code.

Reply:
you are absolutely right, pure laziness on my part.